### PR TITLE
Add tests and assorted fixes and improvements

### PR DESCRIPTION
**Pull request by dashea**
These patches, in brief, do the following:

- Add a test suite (RETR and SIZE only, I didn't do anything with STOR or other commands)
- Some changes to make the FTP adapter behave more like the HTTP-based adapters, so a ftp-ized requests Session can switch between http and ftp URLs more transparently
- Add proxy support, because that's a thing somehow? If you do curl --proxy *proxy* ftp://... it'll do it as an http request, so this change just passes the request to HTTPAdapter
- miscellaneous minor bug fixes

### CODE REVIEW

File: MANIFEST.in [hunk 1]
```diff
@@ -0,0 +1,2 @@
+include test/*.py
```
Comment (Lukasa) [inline MANIFEST.in]: I don't normally see any point in shipping tests on PyPI.

**Comment by Lukasa** [comment]
This is astonishing work, well done! I'm happy to merge it,with one note inline. Also, normally I'd rather use py.test instead of nose for tests: are you open to changing for this pull request?

File: MANIFEST.in (hunk 1 above)
Comment (dashea) [inline MANIFEST.in]: Fair enough. The only issue is that distutils/setuptools picks up test/test_*.py by default, so I just figured I'd include the rest. I can rename the directory to 'tests' instead so it doesn't pick up any of it.

**Comment by dashea** [comment]
Sure, I don't have any particular attachment to nose. I'll look into switching it to py.test tomorrow. A lazy first attempt (copying and pasting the setup.py chunk from the docs) resulted in no tests being run at all, and running py.test from the directory failed when it couldn't find requests_ftp. So I'm guessing something will have be to be lightly rearranged.

**Comment by dashea** [comment]
Changed the tests to py.test, moved "test" to "tests" so that it doesn't get picked up by setup.py sdist, and left setup.py alone as far the test target.

**Comment by Lukasa** [comment]
\o/ It's beautiful! :cake: :sparkles:

**Pull request merged**